src/Model/Component/YamlService.php

```diff
@@ -98,4 +99,44 @@ public static function isLineStartOfArrayWithKeyAndValue($trimmedLine)
     {
         return $trimmedLine !== '-' && self::hasLineDashOnStartOfLine($trimmedLine);
     }
+
+    /**
+     * value starting with key, e.g. 'foo: bar' or '"foo bar": baz'
+     *
+     * @param string $value
+     * @return bool
+     */
+    public static function isKeyInStartOfString($value)
+    {
+        return (bool)preg_match('~^(' . Inline::REGEX_QUOTED_STRING . '|[^ \'"{\[].*?) *:~u', $value);
+    }
+
+    /**
+     * line possibly opening an array, e.g. 'foo:' or '- foo:'
+     *
+     * @param string $trimmedLine
+     * @return bool
+     */
+    public static function isLineOpeningAnArray($trimmedLine)
+    {
+        return (bool)preg_match('~^(- +)*(' . Inline::REGEX_QUOTED_STRING . '|[^ \'"{\[].*?) *:$~u', $trimmedLine);
```

Comment:
these regexes are killers 😄 but I will hope they work as is described 👍

Reply:
I've pretty much used key matching that is used in Symfony `Yaml\Parser`:

```php
preg_match('#^(?P<key>'.Inline::REGEX_QUOTED_STRING.'|[^ \'"\{\[].*?) *\: ...
```

https://github.com/symfony/symfony/blob/288bc24d0362fe87c4eb5c67f89d08f9b93e5807/src/Symfony/Component/Yaml/Parser.php#L191

(from version `v4.3.2`)

Reply:
it's ok, just I write my thinks 😄